pymeasure/instruments/instrument.py

```diff
@@ -63,22 +63,41 @@ class Object(object):
 
         self.get = Object()
 
-        # TODO: Determine case basis for the addition of these methods
-        if includeSCPI:
-            # Basic SCPI commands
-            self.status = self.measurement("*STB?",
-                                           """ Returns the status of the instrument """)
-            self.complete = self.measurement("*OPC?",
-                                             """ TODO: Add this doc """)
-
         self.isShutdown = False
         log.info("Initializing %s." % self.name)
 
+    @property
+    def complete(self):
+        """ This property allows synchronization between a controller and a device. The Operation Complete 
+        query places an ASCII character 1 into the device's Output Queue when all pending
+        selected device operations have been finished.
+        """
+        if self.SCPI:
+            return self.ask("*OPC?").strip()
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    @property
+    def status(self):
+        """ Requests and returns the status byte and Master Summary Status bit. """
+        if self.SCPI:
+            return self.ask("*STB?").strip()
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    @property
+    def options(self):
+        """ Requests and returns the device options installed. """
+        if self.SCPI:
+            return self.ask("*OPT?").strip()
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
     @property
     def id(self):
         """ Requests and returns the identification of the instrument. """
         if self.SCPI:
-            return self.adapter.ask("*IDN?").strip()
+            return self.ask("*IDN?").strip()
         else:
             return "Warning: Property not implemented."
```

Comment:
I think you missed a error message -> NotImplementedError

Reply:
Fixed. Sorry about this.